### ISSUE TITLE
Host network and SELinux type

### ISSUE THREAD

**Issue by ooraini** [cri-o/cri-o#5501]
When running a Pod with `hostNetwork` the container process will always take the privileged `spc_t` type.

This issue was reported in [openshift/origin/issues/25488](https://github.com/openshift/origin/issues/25488). From the comments the relevant file [container_create_linux.go](https://github.com/cri-o/cri-o/blob/main/server/container_create_linux.go) and [commit](https://github.com/cri-o/cri-o/commit/349b85a7f2b97d23fd65490289bf5d66e888e0f9).

It's not clear what is the reason for not allowing it. If possible, can we remove the restriction?

**Comment by saschagrunert** [comment]
@mrunalp can you remember what the reason for https://github.com/cri-o/cri-o/pull/2900 was?

**Comment by haircommander** [comment]
Basically, we assume that if your container has access to hostNetwork, it is sufficiently powerful and trusted to be able to access files on the host. There are many cases where hostNetwork pods assume they can touch files on the host that aren't labeled to be touched by a container, and a pod that could be considered privileged enough gets denied.

What is the case where you want a specific label but the pod also is hostNetwork?

**Comment by ooraini** [comment]
There is case for Ingress controllers to deployed on the host network. Ingress is on the data path for every HTTP request entering the cluster and you would definitely want it be more secure. It's not really a strong case as you have to escape the container(somehow) for SELinux to be relevant(also the case for all other containers).

**Comment by rhatdan** [comment]
That is a bad decision, I know of no reason why a container should run SELinux privileged just because it is in the host network.  PID Host and IPC Host is a different matter.  But the network namespace does not help or hurt SELinux.

**Comment by saschagrunert** [comment]
I fear that we introduce a breaking behavior when we remove it. Even if we make it configurable, at some point we have to turn the defaults to `false` which would then also break on upgrades.

**Comment by rhatdan** [comment]
SELinux does not enforce at the network layer.  So I don't see this as a breaking change.  There is potential that a container does need SELinux to be disabled but not because it is sharing a network namespace.  I believe this should be changed, perhaps wait until a next major release.  But I don't think we should say we can never increase security because we are afraid we might break something.

**Comment by saschagrunert** [comment]
Good, let's give it a try in https://github.com/cri-o/cri-o/pull/5512

**Comment by saschagrunert** [comment]
Closing this one, see https://github.com/cri-o/cri-o/pull/5512#issuecomment-1010762743

**Comment by haircommander** [comment]
cross posting with https://github.com/cri-o/cri-o/pull/5512#issuecomment-1011092477

I just had an idea: what if we added a config field that toggled this behavior? Add field in 1.24, true by default and deprecated by 1.25, dropped field in 1.26. that way, we give users time to update their deployments

**Comment by saschagrunert** [comment]
> I just had an idea: what if we added a config field that toggled this behavior? Add field in 1.24, true by default and deprecated by 1.25, dropped field in 1.26. that way, we give users time to update their deployments

It would not solve the issue that, if we switch the default, it would implicitly break deployments if the user did not take any action before. :thinking:

**Comment by haircommander** [comment]
> It would not solve the issue that, if we switch the default, it would implicitly break deployments if the user did not take any action before. thinking

true, though if we were clear enough in documentation/logging *and* manually tested some of the upstream plugins as part of the deprecation, we could largely mitigate the issues AFAICT

**Comment by saschagrunert** [comment]
This is a good first issue for a new contributor. :+1: 

/help
/milestone 1.24

**Comment by rhatdan** [comment]
If the config gets added, it should default to secure.

**Comment by haircommander** [comment]
this slipped 1.24, maybe we can address in 1.25

**Comment by saschagrunert** [comment]
A friendly reminder that this issue had no activity for 30 days.

**Comment by hasan4791** [comment]
After going through the discussion, this is what we need to do

1. Add a new config field/parameter in crio.conf for 1.25 and default value would be "true"
2. Deprecate the same in 1.26
3. Remove the parameter in future k8s versions

Please correct me.

**Comment by haircommander** [comment]
yeah though start with main/1.27 (by slipping 1.25 and 1.26 we shouldn't backport). So add the field, default to true for 1.27, update it to false for 1.28, possibly remove in the future (or keep it if there's a sustained need)

**Comment by hasan4791** [comment]
/assign
Perfect, assigning myslef

**Comment by hasan4791** [comment]
Kindly suggest, if we need to change the parameter name.
